app/gui/view/component-browser/component-list-panel/icons/src/define_macro.rs

```diff
@@ -81,30 +81,22 @@ macro_rules! define_icons {
 
         impl Id {
             /// Create icon's shape with given size.
-            pub fn create_shape(&self, size: Vector2) -> $crate::entry::icon::Any {
+            pub fn create_shape(&self, size: Vector2) -> $crate::Any {
                 match self {$(
                     Self::$variant => {
                         let view = $name::View::new();
                         view.size.set(size);
-                        let vivid_color_fn = Box::new(f!([view]()
-                            color::Lcha::from(color::Rgba::from(view.vivid_color.get()))
+                        let color_fn = Box::new(f!([view]()
+                            color::Lcha::from(color::Rgba::from(view.color.get()))
                         ));
-                        let dull_color_fn = Box::new(f!([view]()
-                            color::Lcha::from(color::Rgba::from(view.dull_color.get()))
-                        ));
-                        let set_vivid_color_fn = Box::new(f!((c)
-                            view.vivid_color.set(color::Rgba::from(c).into())
-                        ));
-                        let set_dull_color_fn = Box::new(f!((c)
-                            view.dull_color.set(color::Rgba::from(c).into())
+                        let set_color_fn = Box::new(f!((c)
+                            view.color.set(color::Rgba::from(c).into())
```

Comment:
Perhaps a miss from previous PR, but why we just don't keep Attribute in `icon::Any` as before, and only add Lchagetters and setters as normal methods of `icon::Any` instead of those boxed closures?

Reply:
A question to @wdanilo, I just copied the code.

Reply:
@farmaazon this change was dictated by change to shape systems. Either we will have a very generic implementation with a lot of strange constraints, or we can have a simple constraint and this implementation. I've chosen this solution because of simplicity in the code.